src/util.ts

```diff
@@ -14,6 +20,39 @@ export type newtype<A, ID extends string> = A & {
   readonly [sym]: ID;
 };
 
+export function getConfigValue<A>(
+  config: WorkspaceConfiguration,
+  key: string
+): { value: A; target: ConfigurationTarget } | undefined {
+  const value = config.get<A>(key);
+  const { defaultValue, workspaceValue } = config.inspect<A>(key)!;
+  if (value) {
+    const getTarget = () => {
+      if (workspaceValue && workspaceValue !== defaultValue) {
+        return ConfigurationTarget.Workspace;
+      } else {
+        return ConfigurationTarget.Global;
+      }
+    };
+    const target = getTarget();
+    return { value, target };
+  } else if (defaultValue) {
+    return {
+      value: defaultValue,
+      target: ConfigurationTarget.Global,
+    };
+  }
+}
+
+export function metalsDir(target: ConfigurationTarget): string {
+  if (target == ConfigurationTarget.Workspace && workspace.workspaceFolders) {
+    const wsDir = workspace.workspaceFolders[0]?.uri.fsPath;
+    return path.join(wsDir, ".metals");
```

Comment:
@dos65 I changed the directory for the server version data to `.metals` since this would be a more consistent naming. We already have things like `.bloop` etc.

Reply:
Hmm. Technically, it isn't it a problem.
I intentionally named is differently for a case when somebody creates a project inside home dir. I understand that it's weird but who knows, maybe that might be a case for a temporal usage(remote/inside container)